src/lib/pageList.ts

```diff
@@ -1,13 +1,13 @@
-export async function getPages(
-	metaGlob: Array<string>
-): Promise<Array<{ [key: string]: string; filename: string; path: string; title?: string }>> {
+export const getPages = async (metaGlob: Record<string, () => Promise<unknown>>) => {
```

Comment:
Why `unknown` ? Look like we know the final structure (that we use*):
```suggestion
export const getPages = async (metaGlob: Record<string, () => Promise<{metadata: { title: string; layout: string; date: string }}>>) => {
```
It will also remove the need to type the data at lines 4 to 6

*Indeed, it's a partial structure, but we only need to know the part we use